### DISCUSSION TITLE
Failed to compile in regards to 'Component'

### DISCUSSION THREAD

**Discussion by ilarumk** [vercel/community#2861] [Help] [answered]
### Summary

Failed to compile.
--
02:02:39.189 |  
02:02:39.189 | ./pages/_app.tsx:7:8
02:02:39.189 | Type error: 'Component' cannot be used as a JSX component.

Here is the code
import "@/styles/globals.css";
import type { AppProps } from "next/app";

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <main className="scroll-smooth antialiased [font-feature-settings:'ss01']">
      <Component {...pageProps} />
    </main>
  );
}

export default MyApp;

### Steps to Reproduce

I am running on my localhost and working fine, however during vercel deployment I get this error

**Comment by amyegan** [comment]
Please share a [minimal reproduction](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so we can get a better look at what is causing the error  🙏

**Reply by ilarumk** [reply]
https://github.com/ilarumk/adventai

**Comment by ilarumk** [comment]
I updated tsconfig.json to 
https://github.com/ilarumk/adventai/commit/5d2c85e78c4aca43d49caf338f70e0cf7d5a83af
Line 27 to 32

```
    "paths": {
      "react": [
        "./node_modules/@types/react"
      ],
      "@/*": [
        "./*"
      ]
```

it works now, but stuck with a different error now


```
Error: The Edge Function "middleware" is referencing unsupported modules:
--
22:35:09.582 | - next: stream, fs, url
22:35:11.690 | NOW_SANDBOX_WORKER_EDGE_FUNCTION_UNSUPPORTED_MODULES: The Edge Function "middleware" is referencing unsupported modules:	- next: stream, fs, url

```

**Reply by amyegan** [reply, marked as answer]
It looks like you're using some Node.js features that are not supported in the Edge Runtime: https://vercel.com/docs/concepts/functions/edge-middleware/limitations#many-node.js-apis-are-not-available

If you modify the Middleware to stop using the unsupported modules, this error will go away 🙇
